docker-compose-files/compose.sh

```diff
@@ -13,6 +13,57 @@ if [[ $# -eq 0 ]] ; then
     exit 0
 fi
 
+# Version check taken from this fine answer: https://stackoverflow.com/a/4025065/4427375
+# This is used to check if the docker compose version is sufficient.
+vercomp () {
+    if [[ $1 == $2 ]]
+    then
+        return 0
+    fi
+    local IFS=.
+    local i ver1=($1) ver2=($2)
+    # fill empty fields in ver1 with zeros
+    for ((i=${#ver1[@]}; i<${#ver2[@]}; i++))
+    do
+        ver1[i]=0
+    done
+    for ((i=0; i<${#ver1[@]}; i++))
+    do
+        if [[ -z ${ver2[i]} ]]
+        then
+            # fill empty fields in ver2 with zeros
+            ver2[i]=0
+        fi
+        if ((10#${ver1[i]} > 10#${ver2[i]}))
+        then
+            return 1
+        fi
+        if ((10#${ver1[i]} < 10#${ver2[i]}))
+        then
+            return 2
+        fi
+    done
+    return 0
+}
+
+testvercomp () {
+    vercomp $1 $2
+    case $? in
+        0) op='=';;
+        1) op='>';;
+        2) op='<';;
+    esac
+    if [[ $op != $3 ]]
+    then
+        echo "FAIL: Your docker compose version is $1. It must be 2.2.0 or higher!"
+        exit
+    else
+        echo "Pass: Docker compose version is $1."
+    fi
+}
+
+testvercomp $(docker-compose --version | cut -d ' ' -f 4 | sed 's/^v//') 2.2.0 '>'
```

Comment:
Not sure of in-depth analysis, but when I was trying to compose my docker, I faced an error with this line here.
Moving this to line 172 fixed the issue and was able to continue.
@grantcurell

Reply:
Hey there.  
The full fix is to put 2 lines at line 172 and line 173.  (remove set -e on top)  the reason is because set -e will cause script exit  based on the code logic for testvercomp.  This will also fix allow users to view the help page of the utlity.

 testvercomp $(docker-compose --version | cut -d ' ' -f 4 | sed 's/^v//') 2.2.0 '>'
  set -e

Reply:
Absolutely. Just noted the git diff. Was not sure of your id to tag in my comment. Fortunate You pitched in as usual for help.
@windsparks33 thanks for the fix. And the details.